### PR TITLE
Proposed per-project linting

### PR DESCRIPTION
**Pull request by jamielob**
See the README. Comments encouraged!

### CODE REVIEW

File: code_style/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  "rules": {
+    "no-extra-parens": 0, // Interferes with jsx
+    "no-underscore-dangle": 0, // Mongo _id
+    "react/prefer-stateless-function": 1, // We'll choose manually
+    "react/prop-types": 1, // Slows down while prototyping / experimenting
+  },
+  "settings": {
+    "import/resolver": "meteor",
+  }
+}
```
Comment (cfnelson) [inline code_style/.eslintrc]: github wants a \n :)

Comment (jamielob) [inline code_style/.eslintrc]: Ha! Oh the irony.

Comment (jamielob) [inline code_style/.eslintrc]: Done!

**Comment by rdickert** [comment]
I've been using the Airbnb guide recently in my own work, and I'm very in favor of this approach. Airbnb can do the hard work of setting reasonable defaults, which is a timesaver and good for collaborating and onboarding new devs. The deviations here from airbnb are minor but address real issues (esp no-extra-parens, which is a drag in jsx). I'll need to live with it for a while to see if there's anything I would change, but this looks great. Nice work, @jamielob!

**Comment by pauldowman** [comment]
👍  Let's use this. But I think it should replace `dot-eslintrc.json` though. I'd vote for keeping the same name because even though the `.json` extension is optional it helps with syntax highlighting and linting in the editor. Also I had named it with "dot-" instead of "." so that it doesn't show up as a hidden file, but I don't feel strongly about it.

Either way I think we should just have one file, whatever it's named.

**Comment by cfnelson** [comment]
I kind of like having the specific es6 features being stated in the eslint file. It can make a user more aware of what specific features they are using. Otherwise looks good. 
Otherwise 👍 
As we will be using per project eslint, can we make a eslint file for meteor 1.2 projects ?

**Comment by jamielob** [comment]
@cfnelson - AirBnB do a pretty good job of documenting all the different features and rules.   I'll clarify the README.

**Comment by jamielob** [comment]
@pauldowman - I'm also going to leave the .eslintrc named as is for now.   The JSON standard doesn't allow for comments and by adding the .json file extension we get a load of red for having them in there.

**Comment by pauldowman** [comment]
👍  Thanks @jamielob

**Pull request merged**